optuna/storages/rdb/storage.py

```diff
@@ -1059,15 +1067,18 @@ def is_empty(self):
         with self._lock:
             return len(self._finished_trials) == 0
 
-    def cache_trial_if_finished(self, trial):
-        # type: (structs.FrozenTrial) -> None
+    def cache_trial_if_finished(self, trial, deepcopy):
+        # type: (structs.FrozenTrial, bool) -> None
 
         if not self._enabled:
             return
 
         if trial.state.is_finished():
             with self._lock:
-                self._finished_trials[trial._trial_id] = copy.deepcopy(trial)
+                if deepcopy:
+                    trial = copy.deepcopy(trial)
```

Comment:
Considering how this method is called, i.e. with a trial that was just constructed from the DB, we can always create a shallow copy here (and we won't need the `deepcopy` argument for the method).

Reply:
You're right. Thanks!
I'll remove the argument.

Reply:
Fixed: https://github.com/optuna/optuna/pull/696/commits/339af5764727e08640747b03416c4a7741678965

Reply:
Ah, we must change the two calls to this method excluding the argument too.

Reply:
Oops..., sorry. Fixed: https://github.com/optuna/optuna/pull/696/commits/0325b95874595ddbd942d930a35ad26e295fdbf1

Reply:
Thanks for the swift fix!